### PR TITLE
WINC-1010: Upgrade to containerd 1.7

### PR DESCRIPTION
**Pull request by aravindhp**
- Update to openshift/hcsshim@5871d0c
- Update to openshift/containerd@1fbd703
- [build] Update containerd version to v1.7.0
- [internal] Update containerd 1.7 config defaults 

Adding the default configuration generated for review purposes
<details><summary>containerd.exe config default</summary>
<p>
    disabled_plugins = []
    imports = []
    oom_score = 0
    plugin_dir = ""
    required_plugins = []
    root = "C:\\ProgramData\\containerd\\root"
    state = "C:\\ProgramData\\containerd\\state"
    temp = ""
    version = 2
    
    [cgroup]
      path = ""
    
    [debug]
      address = ""
      format = ""
      gid = 0
      level = ""
      uid = 0
    
    [grpc]
      address = "\\\\.\\pipe\\containerd-containerd"
      gid = 0
      max_recv_message_size = 16777216
      max_send_message_size = 16777216
      tcp_address = ""
      tcp_tls_ca = ""
      tcp_tls_cert = ""
      tcp_tls_key = ""
      uid = 0
    
    [metrics]
      address = ""
      grpc_histogram = false
    
    [plugins]
    
      [plugins."io.containerd.gc.v1.scheduler"]
        deletion_threshold = 0
        mutation_threshold = 100
        pause_threshold = 0.02
        schedule_delay = "0s"
        startup_delay = "100ms"
    
      [plugins."io.containerd.grpc.v1.cri"]
        cdi_spec_dirs = []
        device_ownership_from_security_context = false
        disable_apparmor = false
        disable_cgroup = false
        disable_hugetlb_controller = false
        disable_proc_mount = false
        disable_tcp_service = true
        drain_exec_sync_io_timeout = "0s"
        enable_cdi = false
        enable_selinux = false
        enable_tls_streaming = false
        enable_unprivileged_icmp = false
        enable_unprivileged_ports = false
        ignore_image_defined_volumes = false
        image_pull_progress_timeout = "1m0s"
        max_concurrent_downloads = 3
        max_container_log_line_size = 16384
        netns_mounts_under_state_dir = false
        restrict_oom_score_adj = false
        sandbox_image = "registry.k8s.io/pause:3.8"
        selinux_category_range = 0
        stats_collect_period = 10
        stream_idle_timeout = "4h0m0s"
        stream_server_address = "127.0.0.1"
        stream_server_port = "0"
        systemd_cgroup = false
        tolerate_missing_hugetlb_controller = false
        unset_seccomp_profile = ""
    
        [plugins."io.containerd.grpc.v1.cri".cni]
          bin_dir = "C:\\Program Files\\containerd\\cni\\bin"
          conf_dir = "C:\\Program Files\\containerd\\cni\\conf"
          conf_template = ""
          ip_pref = ""
          max_conf_num = 1
          setup_serially = false
    
        [plugins."io.containerd.grpc.v1.cri".containerd]
          default_runtime_name = "runhcs-wcow-process"
          disable_snapshot_annotations = false
          discard_unpacked_layers = false
          ignore_blockio_not_enabled_errors = false
          ignore_rdt_not_enabled_errors = false
          no_pivot = false
          snapshotter = "windows"
    
          [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime]
            base_runtime_spec = ""
            cni_conf_dir = ""
            cni_max_conf_num = 0
            container_annotations = []
            pod_annotations = []
            privileged_without_host_devices = false
            privileged_without_host_devices_all_devices_allowed = false
            runtime_engine = ""
            runtime_path = ""
            runtime_root = ""
            runtime_type = ""
            sandbox_mode = ""
            snapshotter = ""
    
            [plugins."io.containerd.grpc.v1.cri".containerd.default_runtime.options]
    
          [plugins."io.containerd.grpc.v1.cri".containerd.runtimes]
    
            [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor]
              base_runtime_spec = ""
              cni_conf_dir = ""
              cni_max_conf_num = 0
              container_annotations = ["io.microsoft.container.*"]
              pod_annotations = ["io.microsoft.virtualmachine.*"]
              privileged_without_host_devices = false
              privileged_without_host_devices_all_devices_allowed = false
              runtime_engine = ""
              runtime_path = ""
              runtime_root = ""
              runtime_type = "io.containerd.runhcs.v1"
              sandbox_mode = ""
              snapshotter = ""
    
              [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-hypervisor.options]
                SandboxIsolation = 1
                ScaleCpuLimitsToSandbox = true
    
            [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-process]
              base_runtime_spec = ""
              cni_conf_dir = ""
              cni_max_conf_num = 0
              container_annotations = ["io.microsoft.container.*"]
              pod_annotations = []
              privileged_without_host_devices = false
              privileged_without_host_devices_all_devices_allowed = false
              runtime_engine = ""
              runtime_path = ""
              runtime_root = ""
              runtime_type = "io.containerd.runhcs.v1"
              sandbox_mode = ""
              snapshotter = ""
    
              [plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runhcs-wcow-process.options]
    
          [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime]
            base_runtime_spec = ""
            cni_conf_dir = ""
            cni_max_conf_num = 0
            container_annotations = []
            pod_annotations = []
            privileged_without_host_devices = false
            privileged_without_host_devices_all_devices_allowed = false
            runtime_engine = ""
            runtime_path = ""
            runtime_root = ""
            runtime_type = ""
            sandbox_mode = ""
            snapshotter = ""
    
            [plugins."io.containerd.grpc.v1.cri".containerd.untrusted_workload_runtime.options]
    
        [plugins."io.containerd.grpc.v1.cri".image_decryption]
          key_model = "node"
    
        [plugins."io.containerd.grpc.v1.cri".registry]
          config_path = ""
    
          [plugins."io.containerd.grpc.v1.cri".registry.auths]
    
          [plugins."io.containerd.grpc.v1.cri".registry.configs]
    
          [plugins."io.containerd.grpc.v1.cri".registry.headers]
    
          [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
    
        [plugins."io.containerd.grpc.v1.cri".x509_key_pair_streaming]
          tls_cert_file = ""
          tls_key_file = ""
    
      [plugins."io.containerd.internal.v1.opt"]
        path = "C:\\ProgramData\\containerd\\root\\opt"
    
      [plugins."io.containerd.internal.v1.restart"]
        interval = "10s"
    
      [plugins."io.containerd.internal.v1.tracing"]
        sampling_ratio = 1.0
        service_name = "containerd"
    
      [plugins."io.containerd.metadata.v1.bolt"]
        content_sharing_policy = "shared"
    
      [plugins."io.containerd.nri.v1.nri"]
        disable = true
        disable_connections = false
        plugin_config_path = "/etc/nri/conf.d"
        plugin_path = "/opt/nri/plugins"
        plugin_registration_timeout = "5s"
        plugin_request_timeout = "2s"
        socket_path = "/var/run/nri/nri.sock"
    
      [plugins."io.containerd.runtime.v2.task"]
        platforms = ["windows/amd64", "linux/amd64"]
        sched_core = false
    
      [plugins."io.containerd.service.v1.diff-service"]
        default = ["windows", "windows-lcow"]
    
      [plugins."io.containerd.service.v1.tasks-service"]
        blockio_config_file = ""
        rdt_config_file = ""
    
      [plugins."io.containerd.tracing.processor.v1.otlp"]
        endpoint = ""
        insecure = false
        protocol = ""
    
      [plugins."io.containerd.transfer.v1.local"]
    
    [proxy_plugins]
    
    [stream_processors]
    
      [stream_processors."io.containerd.ocicrypt.decoder.v1.tar"]
        accepts = ["application/vnd.oci.image.layer.v1.tar+encrypted"]
        args = ["--decryption-keys-path", "C:\\Program Files\\containerd\\ocicrypt\\keys"]
        env = ["OCICRYPT_KEYPROVIDER_CONFIG=C:\\Program Files\\containerd\\ocicrypt\\ocicrypt_keyprovider.conf"]
        path = "ctd-decoder"
        returns = "application/vnd.oci.image.layer.v1.tar"
    
      [stream_processors."io.containerd.ocicrypt.decoder.v1.tar.gzip"]
        accepts = ["application/vnd.oci.image.layer.v1.tar+gzip+encrypted"]
        args = ["--decryption-keys-path", "C:\\Program Files\\containerd\\ocicrypt\\keys"]
        env = ["OCICRYPT_KEYPROVIDER_CONFIG=C:\\Program Files\\containerd\\ocicrypt\\ocicrypt_keyprovider.conf"]
        path = "ctd-decoder"
        returns = "application/vnd.oci.image.layer.v1.tar+gzip"
    
    [timeouts]
      "io.containerd.timeout.bolt.open" = "0s"
      "io.containerd.timeout.metrics.shimstats" = "2s"
      "io.containerd.timeout.shim.cleanup" = "5s"
      "io.containerd.timeout.shim.load" = "5s"
      "io.containerd.timeout.shim.shutdown" = "3s"
      "io.containerd.timeout.task.state" = "2s"
    
    [ttrpc]
      address = ""
      gid = 0
      uid = 0
</p>
</details>

### CODE REVIEW

**Comment by aravindhp** [comment]
/test gcp-e2e-operator 

One of the control plane nodes failed to come up:
```
level=error msg=Cluster operator authentication Degraded is True with WellKnownReadyController_SyncError: WellKnownReadyControllerDegraded: need at least 3 kube-apiservers, got 2
level=info msg=Cluster operator authentication Progressing is True with WellKnownReadyController_OAuthMetadataNotYetServed: WellKnownReadyControllerProgressing: kube-apiserver oauth endpoint https://10.0.0.5:6443/.well-known/oauth-authorization-server is not yet served and authentication operator keeps waiting (check kube-apiserver operator, and check that instances roll out successfully, which can take several minutes per instance)
level=error msg=Cluster operator authentication Available is False with WellKnown_NotReady: WellKnownAvailable: The well-known endpoint is not yet available: need at least 3 kube-apiservers, got 2
level=info msg=Cluster operator baremetal Disabled is False with : 
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerAvailable is True with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerDegraded is False with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerAvailable is True with AsExpected: Cloud Config Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerDegraded is False with AsExpected: Cloud Config Controller works as expected
level=error msg=Cluster operator control-plane-machine-set Available is False with UnavailableReplicas: Missing 1 available replica(s)
level=info msg=Cluster operator control-plane-machine-set Progressing is True with NeedsUpdateReplicas: Observed 1 replica(s) in need of update
level=info msg=Cluster operator etcd RecentBackup is Unknown with ControllerStarted: The etcd backup controller is starting, and will decide if recent backups are available or if a backup is required
level=info msg=Cluster operator ingress EvaluationConditionsDetected is False with AsExpected: 
level=info msg=Cluster operator insights ClusterTransferAvailable is False with NoClusterTransfer: no available cluster transfer
level=info msg=Cluster operator insights Disabled is False with AsExpected: 
level=info msg=Cluster operator insights SCAAvailable is False with Forbidden: Failed to pull SCA certs from https://api.openshift.com/api/accounts_mgmt/v1/certificates: OCM API https://api.openshift.com/api/accounts_mgmt/v1/certificates returned HTTP 403: {"code":"ACCT-MGMT-11","href":"/api/accounts_mgmt/v1/errors/11","id":"11","kind":"Error","operation_id":"d6d243ba-7093-460a-9120-0b339c96a730","reason":"Account with ID 2DUeKzzTD9ngfsQ6YgkzdJn1jA4 denied access to perform create on Certificate with HTTP call POST /api/accounts_mgmt/v1/certificates"}
level=info msg=Cluster operator network ManagementStateDegraded is False with : 
level=info msg=Use the following commands to gather logs from the cluster
level=info msg=openshift-install gather bootstrap --help
level=error msg=Bootstrap failed to complete: timed out waiting for the condition
level=error msg=Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
level=error msg=Cluster operator authentication Degraded is True with WellKnownReadyController_SyncError: WellKnownReadyControllerDegraded: need at least 3 kube-apiservers, got 2
level=info msg=Cluster operator authentication Progressing is True with WellKnownReadyController_OAuthMetadataNotYetServed: WellKnownReadyControllerProgressing: kube-apiserver oauth endpoint https://10.0.0.5:6443/.well-known/oauth-authorization-server is not yet served and authentication operator keeps waiting (check kube-apiserver operator, and check that instances roll out successfully, which can take several minutes per instance)
level=error msg=Cluster operator authentication Available is False with WellKnown_NotReady: WellKnownAvailable: The well-known endpoint is not yet available: need at least 3 kube-apiservers, got 2
level=info msg=Cluster operator baremetal Disabled is False with : 
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerAvailable is True with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager TrustedCABundleControllerControllerDegraded is False with AsExpected: Trusted CA Bundle Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerAvailable is True with AsExpected: Cloud Config Controller works as expected
level=info msg=Cluster operator cloud-controller-manager CloudConfigControllerDegraded is False with AsExpected: Cloud Config Controller works as expected
level=error msg=Cluster operator control-plane-machine-set Available is False with UnavailableReplicas: Missing 1 available replica(s)
level=info msg=Cluster operator control-plane-machine-set Progressing is True with NeedsUpdateReplicas: Observed 1 replica(s) in need of update
level=info msg=Cluster operator etcd RecentBackup is Unknown with ControllerStarted: The etcd backup controller is starting, and will decide if recent backups are available or if a backup is required
level=info msg=Cluster operator ingress EvaluationConditionsDetected is False with AsExpected: 
level=info msg=Cluster operator insights ClusterTransferAvailable is False with NoClusterTransfer: no available cluster transfer
level=info msg=Cluster operator insights Disabled is False with AsExpected: 
level=info msg=Cluster operator insights SCAAvailable is False with Forbidden: Failed to pull SCA certs from https://api.openshift.com/api/accounts_mgmt/v1/certificates: OCM API https://api.openshift.com/api/accounts_mgmt/v1/certificates returned HTTP 403: {"code":"ACCT-MGMT-11","href":"/api/accounts_mgmt/v1/errors/11","id":"11","kind":"Error","operation_id":"d6d243ba-7093-460a-9120-0b339c96a730","reason":"Account with ID 2DUeKzzTD9ngfsQ6YgkzdJn1jA4 denied access to perform create on Certificate with HTTP call POST /api/accounts_mgmt/v1/certificates"}
level=info msg=Cluster operator network ManagementStateDegraded is False with : 
level=error msg=Bootstrap failed to complete: timed out waiting for the condition
level=error msg=Failed to wait for bootstrapping to complete. This error usually happens when there is a problem with control plane hosts that prevents the control plane operators from creating the control plane.
level=warning msg=The bootstrap machine is unable to resolve API and/or API-Int Server URLs
level=info
level=info msg=Bootstrap gather logs captured here "/tmp/installer/log-bundle-20230322224316.tar.gz"
Installer exit with code 5
```

File: pkg/internal/containerd_conf.toml [hunk 1]
```diff
@@ -68,12 +76,16 @@ version = 2
       bin_dir = "C:\\k\\cni"
       conf_dir = "C:\\k\\cni\\config"
       conf_template = ""
+      ip_pref = "ipv4"
```
Comment (saifshaikh48) [inline pkg/internal/containerd_conf.toml:79]: was this set manually to ipv4? or is this also generated

**Review by saifshaikh48 — COMMENTED** [review]
lgtm but needs rebase

File: pkg/internal/containerd_conf.toml (hunk 1 above)
Comment (aravindhp) [reply pkg/internal/containerd_conf.toml:79]: It was set to `""`. `""` and `ipv4` are [synonymous](https://github.com/containerd/containerd/blob/992067339161f7e649a23db160943c8b4709c790/docs/cri/config.md?plain=1#L412). I modified it to `ipv4` to make it clear.

**Comment by mansikulkarni96** [comment]
LGTM

**Comment by aravindhp** [comment]
VMC vSphere issue with OVAs has been resolved

/retest-required

**Pull request merged**